packages/gatsby-theme-newrelic/src/components/EditPageButton.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import Button from './Button';
+import Icon from './Icon';
+import Link from './Link';
+import useInstrumentedHandler from '../hooks/useInstrumentedHandler';
+import useThemeTranslation from '../hooks/useThemeTranslation';
+import { graphql, useStaticQuery } from 'gatsby';
+
+const EditPageButton = ({
+  fileRelativePath,
+  onClick,
+  instrumentation,
+  ...props
+}) => {
+  const { t } = useThemeTranslation();
+  const handleClick = useInstrumentedHandler(onClick, {
+    actionName: 'editThisPage_click',
+    component: instrumentation?.component,
+  });
+
+  const {
+    site: {
+      siteMetadata: { repository, branch },
+    },
+  } = useStaticQuery(graphql`
+    query {
+      site {
+        siteMetadata {
+          repository
+          branch
+        }
+      }
+    }
+  `);
+
+  return (
+    <Button
+      {...props}
+      as={Link}
+      to={`${repository}/blob/${branch}/${fileRelativePath || ''}`}
+      onClick={handleClick}
+    >
+      <Icon
+        name="fe-edit"
+        css={css`
+          margin-right: 0.5rem;
+        `}
+      />
+      {t('github.editPage')}
+    </Button>
+  );
+};
+
+EditPageButton.propTypes = {
+  fileRelativePath: PropTypes.string.isRequired,
+  onClick: PropTypes.func,
+  instrumentation: PropTypes.shape({
+    component: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+export default EditPageButton;
```

Comment:
How do you feel about importing `Button` and exporting it's sizes, variants, etc? If feels odd that we need to bring in both components just to gain access to those properties.

I don't feel strongly about it, just curious.

Reply:
I noodled on this a bit and went back and forth. Why I didn't go the route of re-exporting is due to the abstractions in a previous PR regarding the `GitHubIssueButton` and `CreateIssueButton`. `CreateIssueButton` wraps `GitHubIssueButton`, which wraps `Button`. `GitHubIssueButton` is a button which adds the ability to link to an issue and provide a template, where `CreateIssueButton` is a very specific `GitHubIssueButton`.

I originally copied over the properties, but it started feeling pretty ridiculous, and very error prone.

```js
// GitHubButton.js
import Button from './Button'

const GitHubButton = () => {...}

GitHubButton.VARIANT = Button.VARIANT
GitHubButton.SIZE = Button.SIZE
```

```js
// CreateIssueButton.js
import GitHubButton from './GitHubButton'

const CreateIssueButton = () => {...}

CreateIssueButton.VARIANT = GitHubIssueButton.VARIANT
CreateIssueButton.SIZE = GitHubIssueButton.SIZE
```

Those `variant` and `size` props are just forwarded to the underlying `Button`. While this doesn't seem too crazy, where I think this gets error prone is when we need new enums or new wrappers around the button. Having to remember to copy these properties each time we create a button abstraction can be easy to forget.

I made the tradeoff that rather than setting the static properties, I'd just let the caller import button to use the static properties.

Thoughts?

Reply:
Yeah, I can see there being a couple of trade-offs.

Maybe it's the lesser evil to import `<EditPageButton />` and `Button` (like you have here - loosely coupled) than to import / export (proposed - tightly coupled).

Looks like I'm also noodling on it. Let's keep it as-is.